### PR TITLE
compare: add support for a wider PNG range

### PR DESCRIPTION
**Pull request by MrSmile**
I've tried to trigger all necessary colorspace conversions. It works on several samples that I created, but it certainly not a full set of possible PNG variants. So, more testing is welcome, especially with different paletted versions.

### CODE REVIEW

File: compare/image.c [hunk 1]
```diff
@@ -60,15 +60,20 @@ bool read_png(const char *path, Image16 *img)
...
-    if (w > 0xFFFF || h > 0xFFFF || type != PNG_COLOR_TYPE_RGBA) {
+    if (w > 0xFFFF || h > 0xFFFF) {
         png_destroy_read_struct(&png, &info, NULL);
         fclose(fp);
         return false;
     }
 
+    if(!(type & PNG_COLOR_MASK_COLOR))png_set_gray_to_rgb(png);
+    if((type & PNG_COLOR_MASK_PALETTE) || a)png_set_expand(png);
+    if(!a && !(type & PNG_COLOR_MASK_ALPHA))png_set_filler(png, -1, PNG_FILLER_AFTER);
```
Comment (rcombs) [inline compare/image.c:75]: Whitespace on all these lines (preferably the conditionalized statements on separate lines from the conditions)

Comment (MrSmile) [reply compare/image.c]: Done.

**Review by TheOneric — APPROVED** [review]
haven’t tested decoding samples, but code seems good

**Comment by TheOneric** [comment]
Here are some tests with 8bpc RGB, 8bpc GRAY+ALPHA and palletised (with alpha). All are accepted as identical to `compare`’s own output. Notably when palettising with GIMP the exported result appear to have the green box shifted by 1px, but palletising with `optipng` gave an identical result.

[pngfmt_test.zip](https://github.com/user-attachments/files/15879766/pngfmt_test.zip)

**Comment by TheOneric** [comment]
@MrSmile: do any more formats need testing, or is it now good for merge?

**Comment by MrSmile** [comment]
Rebased.

> do any more formats need testing

Both my and yours samples are from GIMP, so I feel that's not fully representative, especially in tRNS case. I've tried `pngcrush` at a couple of samples, but failed to produce a new color format with it (maybe I don't know how to use it correctly). Well, we can always merge as it is and fix errors afterwards, it's debug functionality anyway.

**Comment by TheOneric** [comment]
> Both my and yours samples are from GIMP, so I feel that's not fully representative, especially in tRNS case.

My palletised alpha case was generated by `optipng` *(since GIMPs output was shifted for some reason judging by imagemagicks diff view)*. `optipng`’s inpout was an RGBA image from GIMP, but afaik this shouldn't matter.

**Pull request merged**